src/data/markdown/translated-guides/en/02 Using k6/14 Scenarios/00 Concepts/02 Arrival-rate VU allocation.md

```diff
@@ -112,3 +112,13 @@ Some of the times it might make sense to use `maxVUs` include:
 - To determine necessary allocation in first-time tests
 - To add a little "cushion" to the pre-allocated VUs that you expect the test needs
 - In huge, highly distributed tests, in which you need to carefully scale load generators as you increment VUs.
+
+## Arrival-rate executors cycle over all VUs
+
+Given enough time, k6 cycles over all VUs initialized, even if only a small fraction of them are needed to reach the target rate.
+
+This fact has implications for the [execution context variables](/docs/using-k6/execution-context-variables/).
+For example, if your arrival-rate scenario allocates 15 vus and runs long enough,
+then the value of `{exec.vu.iterationInScenario}` will reach `15` at some points,
+even if k6 never requires that number of VUs to reach the target iteration.
+
```

Comment:
```suggestion
then the value of `{exec.vu.iterationInScenario}` will reach `15` at some points,
even if k6 never requires that number of VUs to reach the target iteration *at the same time*.

```

Reply:
Otherwise I can read this as even if I only had to to 10 iterations (1iter/s for 10s) but got 15 VUs I will still use all 15 VUs ... somehow

Reply:
Hmm I'm a little lost at both yours and mine.

Maybe:

> even if the executor never requires all 15 VUs to reach the target rate.

WDYT?

Reply:
I mean it is better as *rate* is the key word. My problem is that somebody who doesn't understand this all that well likely will be just as confused. 

IMO it is better to be more explicit in what is that we are trying to say when it is confusing to begin with. 

In this case for me the key parts are:
1. k6 will use at most as many VUs *at a time* to hit the provided rate
2. this VUs will not necessarily be the first N VUs *only* even if only N or less VU are needed at any given moment.

I am not certain guaranteeing that it will go through all the VUs is a good idea - this is an implementation detail that likely is never necessary to be known.